src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py

```diff
@@ -656,21 +657,137 @@ def evaluate(self, dataset, metric = 'auto', verbose = True):
+        +----+-------+-------------------+
+        | 0  |   4   |   0.995623886585  |
+        | 0  |   9   |  0.0038311756216  |
+        | 0  |   7   | 0.000301006948575 |
+        | 1  |   1   |   0.928708016872  |
+        | 1  |   3   |  0.0440889261663  |
+        | 1  |   2   |  0.0176190119237  |
+        | 2  |   3   |   0.996967732906  |
+        | 2  |   2   |  0.00151345680933 |
+        | 2  |   7   | 0.000637513934635 |
+        | 3  |   1   |   0.998070061207  |
+        | .. |  ...  |        ...        |
+        +----+-------+-------------------+
+        [35688 rows x 3 columns]
+        """
+        _tkutl._check_categorical_option_type("output_type", output_type, 
+            ["probability", "rank"])
+        int_type_error_string = (lambda var_name: ("'{var_name}' must be " 
+            + "an integer >= 1").format(var_name=var_name))
+        non_neg_range_error_string = (lambda var_name: ("'{var_name}' must be " 
+            + ">= 1").format(var_name=var_name))
+        if not isinstance(k, int): raise TypeError(int_type_error_string("k"))
+        if k <= 0: raise ValueError(non_neg_range_error_string("k"))
+        if batch_size is not None and not isinstance(batch_size, int):
+            raise TypeError(int_type_error_string("batch_size"))
+        if batch_size is not None and batch_size < 1:
+            raise ValueError(non_neg_range_error_string("batch_size"))
+
+        prob_vector = self.predict(
+            dataset, output_type='probability_vector', batch_size=batch_size)
+
+        classes = self.classes
+        if output_type == 'probability':
+            results = prob_vector.apply(lambda p: [
+                            {'class': classes[i], 'probability': p[i]}
+                            for i in reversed(_np.argsort(p)[-k:])]
+                        )
+        else:
+            assert(output_type == 'rank')
+            def convert_probability_vector_to_ranks(p):
+                rank, ranks_list = 0, []
+                for index in reversed(_np.argsort(p)[-k:]):
+                    ranks_list.append({
+                        'class': classes[index], 'rank': rank
+                        })
+                    rank = (rank + 1) % k
+                return ranks_list
+            results = prob_vector.apply(convert_probability_vector_to_ranks)
+
+        results = _tc.SFrame({'X': results})
+        results = results.add_row_number()
+        results = results.stack('X', new_column_name='X')
+        results = results.unpack('X', column_name_prefix='')
+        return results
+        
+
+    def predict(self, data, output_type='class', batch_size=None, verbose=True):
         """
         Predict on an SFrame or SArray of drawings, or on a single drawing.
 
         Parameters
         ----------
         data : SFrame | SArray | tc.Image | list
-            The image(s) on which to perform drawing classification.
+            The drawing(s) on which to perform drawing classification.
```

Comment:
It would be best to be consistent about image(s)/drawing(s). It looks like the other docstring in this commit uses "image".

Reply:
Done.